packages/textlint/src/createLinter.ts

```diff
@@ -113,13 +133,36 @@ export const createLinter = (options: CreateLinterOptions) => {
                 return kernel.fixText(fileContent, kernelOptions);
             });
         },
+        /**
+         * Lint text and fix it
+         * Note: fixText does not respect ignore file
+         * You can detect the file path is ignored or not by `scanFilePath()`
+         * @param text
+         * @param filePath
+         */
         async fixText(text: string, filePath: string): Promise<TextlintFixResult> {
             const kernelOptions = {
                 ext: path.extname(filePath),
                 filePath,
                 ...baseOptions
             };
             return kernel.fixText(text, kernelOptions);
+        },
+        /**
+         * Scan file path and return scan result
+         * If you want to know the file is ignored by ignore file, use this function
+         * Return { status "ok" | "ignored" | "error" } object:
+         * - ok: found file and allow to lint/fix
+         * - ignored: found file, and it is ignored by ignore file
+         * - error: not found file
+         * @param filePath
+         * @returns {Promise<ScanFilePathResult>}
+         */
+        async scanFilePath(filePath: string): Promise<ScanFilePathResult> {
+            return scanFilePath(filePath, {
+                cwd,
+                ignoreFilePath: options.ignoreFilePath
+            });
         }
```

Comment:
This PR Add `linter.scanFilePath()` API 
If you want to know the file is ignored by ignore file, you can use this function.

```js
const scanResult = await linter.scanFilePath(filePath);
console.log(scanResult.status);// => ok or ignored or error
```

cc @frozenbonito